src/App.tsx

```diff
@@ -1,29 +1,77 @@
-import { useSelector } from 'react-redux';
+/* eslint-disable max-len */
+import React, {
+  useEffect,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
-import { Start } from './components/Start';
-import { Finish } from './components/Finish';
+import { useDispatch, useSelector } from 'react-redux';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { actions as todosAction } from './features/todos';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { RootState } from './app/store';
+import { TodoModal } from './components/TodoModal';
+import { Loader } from './components/Loader';
 
-import { selectors } from './store';
+export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
 
-export const App = () => {
-  // `useSelector` connects our component to the Redux store
-  // and rerenders it after every dispatched action
-  const loading = useSelector(selectors.isLoading);
+  const dispatch = useDispatch();
+  const todos: Todo[] = useSelector((state: RootState) => {
+    if (state.filter.status === 'active') {
+      return state.todos.filter((t: Todo) => (t.title.toLowerCase()
+        .includes(state.filter.query.toLowerCase())) && t.completed === false);
+    }
 
-  // we do not call a selector with (), just pass a link to it
-  const message = useSelector(selectors.getMessage) || 'Ready!';
+    if (state.filter.status === 'completed') {
+      return state.todos.filter((t: Todo) => (t.title.toLowerCase()
+        .includes(state.filter.query.toLowerCase())) && t.completed === true);
+    }
+
+    return state.todos.filter((t: Todo) => t.title.toLowerCase()
+      .includes(state.filter.query.toLowerCase()));
+  });
+
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
```

Comment:
Create a selector for each redux store value, don't write them inside components, just call needed selectors.
Example `const loading = useSelector(selectors.isLoading);`

Fix here and in all other places.